src/router.js

```diff
@@ -158,7 +158,17 @@ export function Router(props) {
 
 	let pathRoute, defaultRoute, matchProps;
 	toChildArray(props.children).some((/** @type {VNode<any>} */ vnode) => {
-		const matches = exec(rest, vnode.props.path, (matchProps = { ...vnode.props, path: rest, query, params, rest: '' }));
+		const matches = exec(
+			rest,
+			vnode.props.path,
+			(matchProps = {
+				...vnode.props,
+				path: rest,
+				query,
+				params: Object.assign({}, params),
```

Comment:
This `Object.assign()` is the only change, rest is just formatting as that line was getting rather long & hard to read.